drivers/crypto/crypto_it8xxx2_sha.c

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (c) 2023 ITE Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ite_it8xxx2_sha
+
+#include <zephyr/kernel.h>
+#include <zephyr/crypto/crypto.h>
+#include <zephyr/sys/byteorder.h>
+#include <chip_chipregs.h>
+#include <errno.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(sha_it8xxx2, CONFIG_CRYPTO_LOG_LEVEL);
+
+BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 1,
+	     "unsupported sha instance");
+
+#define IT8XXX2_SHA_REGS_BASE \
+	((struct it8xxx2_sha_regs *)DT_REG_ADDR(DT_NODELABEL(sha0)))
+
+#define IT8XXX2_SHA_START_SHA256 BIT(1)
+
+#define SHA_SHA256_HASH_LEN        32
+#define SHA_SHA256_BLOCK_LEN       64
+#define SHA_SHA256_K_LEN           256
+#define SHA_SHA256_HASH_LEN_WORDS  (SHA_SHA256_HASH_LEN / sizeof(uint32_t))
+#define SHA_SHA256_BLOCK_LEN_WORDS (SHA_SHA256_BLOCK_LEN / sizeof(uint32_t))
+#define SHA_SHA256_K_LEN_WORDS     (SHA_SHA256_K_LEN / sizeof(uint32_t))
+
+/*
+ * This struct is used by the hardware and must be stored in RAM first 4k-byte
+ * and aligned on a 256-byte boundary.
+ */
+struct chip_sha256_ctx {
+	union {
+		/* W[0] ~ W[15] */
+		uint32_t w_sha[SHA_SHA256_BLOCK_LEN_WORDS];
+		uint8_t w_input[SHA_SHA256_BLOCK_LEN];
+	};
+	/* reserved */
+	uint32_t reserved1[8];
+	/* H[0] ~ H[7] */
+	uint32_t h[SHA_SHA256_HASH_LEN_WORDS];
+	/* reserved */
+	uint32_t reserved2[30];
+	uint32_t w_input_index;
+	uint32_t total_len;
+	/* K[0] ~ K[63] */
+	uint32_t k[SHA_SHA256_K_LEN_WORDS];
+} __aligned(256);
+
+__attribute__((section(".__sha256_ram_block"))) __attribute__((used))
```

Comment:
There is a macro for that, `Z_GENERIC_SECTION`

Reply:
Done